mlflow/models/model.py

```diff
@@ -996,14 +996,25 @@ def update_model_requirements(
     old_conda_reqs = _get_requirements_from_file(conda_yaml_path)
     old_requirements_reqs = _get_requirements_from_file(requirements_txt_path)
 
+    requirements = []
+    invalid_requirements = {}
+    for s in requirement_list:
+        try:
+            requirements.append(Requirement(s.strip().lower()))
+        except InvalidRequirement as e:
+            invalid_requirements[s] = e
```

Comment:
What does the error message look like? If it just says "invalid requirement", we don't need to include it in the message.

Reply:
```
>>> invalid_reqs = ["mlflow=", "abc>", "~package"]
>>> invalid_requirements = {}
>>> for req in invalid_reqs:
...     try:
...             Requirement(req)
...     except Exception as e:
...             invalid_requirements[req] = e
... 
>>> invalid_requirements
{'mlflow=': InvalidRequirement('Expected end or semicolon (after name and no valid version specifier)\n    mlflow=\n          ^'), 'abc>': InvalidRequirement('Expected end or semicolon (after name and no valid version specifier)\n    abc>\n       ^'), '~package': InvalidRequirement('Expected package name at the start of dependency specifier\n    ~package\n    ^')}
```
The error message seems useful